lib/serverengine/supervisor.rb

```diff
@@ -194,20 +239,29 @@ def try_join
     end
 
     def start_server
-      s = create_server(logger)
-      @last_start_time = Time.now
+      unless ServerEngine.windows?
+        s = create_server(logger)
+        @last_start_time = Time.now
 
-      begin
-        m = @pm.fork do
-          $0 = @server_process_name if @server_process_name
-          s.install_signal_handlers
+        begin
+          m = @pm.fork do
+            $0 = @server_process_name if @server_process_name
+            s.install_signal_handlers
+
+            s.main
+          end
 
-          s.main
+          return m
+        ensure
+          s.after_start
         end
+      else # if ServerEngine.windows?
+        inpipe, @command_pipe = IO.pipe
+        @last_start_time = Time.now
+        m = @pm.spawn(*Array(config[:windows_daemon_cmdline]), in: inpipe)
```

Comment:
My intention is to support it on Unix, to write same script using ServerEngine on both Unix and Windows.